Realm/Realm.Unity/Editor/Extensions/MethodReferenceExtensions.cs

```diff
@@ -49,4 +49,14 @@ public static MethodReference MakeHostInstanceGeneric(this MethodReference @this
 
         return reference;
     }
+
+    public static bool ConstructsType(this MethodReference @this, TypeReference type)
+    {
+        if (@this is null)
+        {
+            return false;
+        }
```

Comment:
Let's remove this check. If this method is called with `null` @this, it'd be a bug, so better let it go through and throw a NRE at the return. That will 1) give us behavior almost identical to the behavior of calling a regular method on a `MethodReference` and 2) allow us to use an expression for the method (which I love 😄).